### ISSUE TITLE
How to optimise speed with AMD on Windows

### ISSUE THREAD

**Issue by InCrIpTiOn** [comfyanonymous/ComfyUI#1204]
I have a dual boot system Ubuntu/Win11. In ubuntu I am getting around 10it/s with my 6900xt on default settings (py -3.10 main.py). On Windows using directml I get 1it/s, usually less, using (py -3.10 main.py --directml). I've tried vram settings, cross attention, dont upcast, but none seem to improve it, any tips?

**Comment by NeedsMoar** [comment]
directml is generally just terrible without using ONNX-optimized models (it's actually pretty damn fast then) and hasn't been updated in a long time.  I think one of the node add-ons for Comfy allows ONNX loading if comfy itself doesn't already so it might be worth looking into that and the onnx runtime directml backend.  I was getting speeds nearly as fast as Shark when testing that, and Shark is the fastest thing there is right now (with lots of limitations).    

Lots of the command line options don't actually do anything with directml enabled (it forces a certain vram model and I'm pretty sure doesn't work with --force-fp16.)  The attention modes are mostly good for getting it to not OOM on larger models but it's still going to allocate all vram.   

AMD has ported RoCM + HIP to Windows and has an SDK & driver release out,  and is prioritizing the MIOpen port (it's already building apparently, but on hold while somebody is on vacation) so the PyTorch people can implement the backend and everybody can just use that.  Most of the UIs are waiting on that.  Speeds should be similar to linux once that's done.     It doesn't do a very good job of exposing memory management APIs (it might not at all for python, there's no documentation on the one function that exists which just returns an array of zeros whose length roughly corresponds to total VRAM in GB / 2 from my testing but can't determine available...  so optimize speed by waiting, in this case.  :P

**Comment by InCrIpTiOn** [comment]
That's right, thank you. It seems like I must convert the models manually to onnx and load them that way, which is a real pain. I hope pytorch are working on this

**Comment by NeedsMoar** [comment]
Some models on huggingface are available as ONNX since diffusers supports it so it's worth checking there first for the more standard models (they'll show up in the dropdown to select the branch of a repo along with any other alternates), although in my experience conversion & optimization w/ Olive is a fairly quick process.

**Comment by RafaelEstevamReis** [comment]
ROCm finally supports windows
https://rocm.docs.amd.com/en/latest/release/windows_support.html

There is any way to use it?

**Comment by InCrIpTiOn** [comment]
@RafaelEstevamReis No, MIOpen, then Pytorch need to add support, see https://github.com/pytorch/pytorch/issues/106161

**Comment by InCrIpTiOn** [comment]
also closing, for best speeds use comfyui on linux 👍

**Comment by InCrIpTiOn** [comment]
also closing, for best speeds use comfyui on linux 👍

**Comment by Andyholm** [comment]
How would I load an onnx model in comfy? I've converted 1.5 to onnx just to test
